### PR TITLE
ISPN-6359 DistTopKeyTest.testPut random failures

### PR DESCRIPTION
**Pull request by pruivo**
https://issues.jboss.org/browse/ISPN-6359

The first commit is only cleanups. no change in the tests.

### CODE REVIEW

File: extended-statistics/src/test/java/org/infinispan/stats/BaseClusterTopKeyTest.java [hunk 1]
```diff
@@ -27,101 +33,111 @@
...
       this.clusterSize = clusterSize;
-      this.cleanup = CleanupPhase.AFTER_METHOD;
    }
 
-   public void testPut() {
-      resetStreamSummary(cache(0));
-      resetStreamSummary(cache(1));
+   @BeforeMethod(alwaysRun = true)
+   public void resetBeforeTest() {
+      caches().forEach(cache -> getTopKey(cache).resetStatistics());
```
Comment (danberindei) [inline extended-statistics/src/test/java/org/infinispan/stats/BaseClusterTopKeyTest.java]: I don't think this is necessary with `@CleanupAfterTest`.

File: extended-statistics/src/test/java/org/infinispan/stats/BaseClusterTopKeyTest.java [hunk 2]
```diff
@@ -325,40 +326,25 @@ public Object visitPrepareCommand(TxInvocationContext ctx, PrepareCommand comman
...
-      public final void run() {
-         try {
-            innerRun();
-         } catch (Throwable throwable) {
-            this.throwable = throwable;
-         }
-      }
-
-      protected abstract void innerRun() throws Throwable;
+   private static String generateKey(Method method, int index) {
```
Comment (danberindei) [inline extended-statistics/src/test/java/org/infinispan/stats/BaseClusterTopKeyTest.java]: You could use `TestingUtil.k(Method, int)`.

Comment (pruivo) [inline extended-statistics/src/test/java/org/infinispan/stats/BaseClusterTopKeyTest.java]: that is a good point :) I'll update

**Comment by pruivo** [comment]
updated!

File: extended-statistics/src/test/java/org/infinispan/stats/BaseClusterTopKeyTest.java [hunk 3]
```diff
@@ -27,101 +33,111 @@
...
+
+      cache(0).put(key1, "value1");
+      assertNoLocks(key1);
 
-      cache(0).put("key1", "value1");
-      cache(0).put("key2", "value2");
-      cache(1).put("key1", "value3");
-      cache(1).put("key2", "value4");
+      cache(0).put(key2, "value2");
+      assertNoLocks(key2);
```
Comment (danberindei) [inline extended-statistics/src/test/java/org/infinispan/stats/BaseClusterTopKeyTest.java]: It should be faster if the `assertNoLock()` were grouped.

**Comment by pruivo** [comment]
updated again @danberindei !

**Comment by danberindei** [comment]
Integrated, thanks Pedro!

**Pull request closed without merging**